pkg/apic/provisioning/schemabuilder_test.go

```diff
@@ -47,4 +48,57 @@ func TestSubscriptionSchemaBuilderSetters(t *testing.T) {
 		Build()
 
 	assert.NotNil(t, err)
+
+	// set property order - property order takes precedence
+	_, err = NewSchemaBuilder().
+		SetName("name").
+		AddUniqueKey("key").
+		SetPropertyOrder([]string{"name3", "name2", "name1"}).
+		AddProperty(NewSchemaPropertyBuilder().
+			SetName("name1").
+			SetDescription("description").
+			SetRequired().
+			IsString().
+			SetEnumValues([]string{"a", "b", "c"})).
+		AddProperty(NewSchemaPropertyBuilder().
+			SetName("name2").
+			SetDescription("description").
+			SetRequired().
+			IsString().
+			SetEnumValues([]string{"a", "b", "c"})).
+		AddProperty(NewSchemaPropertyBuilder().
+			SetName("name3").
+			SetDescription("description").
+			SetRequired().
+			IsString().
+			SetEnumValues([]string{"a", "b", "c"})).
+		Build()
+
+	assert.Nil(t, err)
```

Comment:
Are you doing validations elsewhere that things were set properly?

Reply:
not really.. i just did a "test" for the reverse order of properties that were not added but property names in the property order.  But, for either of those scenarios, we don't throw any errors, just warns.  So, there is no assertNotNil for error right? Unless I'm misunderstanding some j-ish.

Reply:
I was wondering if you could check the property order in the returned object, currently _, for the correct value.

Reply:
ah...gotcha!  can do.  thanks!

Reply:
Is it worth checking? I see no other validations check anything but for err.

Reply:
agreed

Reply:
Created some checks and balances in regards to property order